tvb_framework/tvb/interfaces/rest/bids_monitoring/README.md

```diff
@@ -0,0 +1,58 @@
+## BIDS data builder and directory monitor
+
+This module contains helper classes which can build BIDS datasets and import them into TVB projects. Along with this, the module also provides a way to monitor a BIDS directory and build datasets accordingly when new files are added.
+
+### BIDSDataBuilder
+
+- Class for building BIDS dataset by providing 1) a set of json files and BIDS root directory or 2) a datatype from the subject directory (e.g. net, ts, coord, spatial)
+- Contains utils for finding all dependencies of a subject json file.
+- Produces a zip file.
+- Takes following args
+   - BIDS_ROOT_DIR (reqd.)- BIDS root dir from which dataset will be created
+   - BIDS_DATA_TO_IMPORT - Accepts BIDSImporter net, ts, coord, spatial token  to build subject type specific dataset For e.g. if BIDSImporter.TS_TOKEN is provided then, it'll build a dataset containing only TS data and their dependencies
+   - INITIAL_JSON_FILES - A set of initial json files with which bids dataset to be built. This is used by BIDSDirWatcher
+
+### BIDSDirWatcher
+
+- Class for monitoring a BIDS directory and also builds datasets accordingly on new files
+- Runs two threads in parallel for 1) observing a directory and 2) building/importing dataset into TVB
+- Importing thread runs on a fixed interval (configurable), which also acts as a buffer time when large files are added
+- Contains utils for observing specific (subjects) directories changes only
+- Takes following args
+    - DIRECTORY_TO_WATCH (reqd.)- BIDS root dir on which is to be monitored
+    - UPLOAD_TRIGGER_INTERVAL - Importer thread interval in seconds
+    - IMPORT_DATA_IN_TVB - A flag for only creating dataset and storing and not importing into TVB projects
+    - TVB_PROJECT_ID - ID of TVB project on which dataset is to be imported
+
+
+[launch_bids_monitor.py](tvb_framework/tvb/interfaces/rest/bids_monitoring/launch_bids_monitor.py) contains sample code and it also accepts command line arguments. To start monitoring a bids directory run below command
+
+```sh
+   $ python launch_bids_monitor.py --rest-url=http://localhost:9090 --bids-dir=user/doc/BIDS_SAMPLE
+```
+
+where  `-rest-url` is the url on which TVB rest server is running and `-bids-dir` is the BIDS root directory which is to be monitored for new files
+
```

Comment:
Would be worth mentioning here that the user needs to add new files under the BIDS subject directories in order for this module to trigger an import into TVB.

Reply:
sure